rust/libnewsboat/src/utils.rs

```diff
@@ -858,21 +858,33 @@ pub fn strip_comments(line: &str) -> &str {
     &line[0..first_pound_chr_idx]
 }
 
+/// The result of executing `extract_filter()`.
+pub struct FilterUrlParts {
+    /// "~/bin/foo.sh" in "filter:~/bin/foo.sh:https://example.com/news.atom"
+    pub script_name: String,
+
+    /// "https://example.com/news.atom" in "filter:~/bin/foo.sh:https://example.com/news.atom"
+    pub url: String,
+}
+
 /// Extract filter and url from line separated by ':'.
-pub fn extract_filter(line: &str) -> (&str, &str) {
+pub fn extract_filter(line: &str) -> FilterUrlParts {
     debug_assert!(line.starts_with("filter:"));
     // line must start with "filter:"
     let line = line.get("filter:".len()..).unwrap();
-    let (filter, url) = line.split_at(line.find(':').unwrap_or(0));
+    let (script_name, url) = line.split_at(line.find(':').unwrap_or(0));
     let url = url.get(1..).unwrap_or("");
     log!(
         Level::Debug,
         "utils::extract_filter: {} -> filter: {} url: {}",
```

Comment:
It's just a debug log, but for consistency the "filter" could be renamed to "script name", too.